geotrek/common/mixins/models.py

```diff
@@ -416,3 +433,36 @@ def add_property(cls, name, func, verbose_name):
             raise AttributeError("%s has already an attribute %s" % (cls, name))
         setattr(cls, name, property(func))
         setattr(cls, '%s_verbose_name' % name, verbose_name)
+
+
+class DuplicateModelMixin(CheckBoxActionMixin):
+    def duplicate(self):
+        """
+        Duplicate all related objects of obj setting
+        field to value. If one of the duplicate
+        objects has an FK to another duplicate object
+        does not update that as well.
+        """
+        from geotrek.core.models import Topology
+
+        collector = Collector(using='default')
+        collector.collect([self])
+        collector.sort()
+        related_models = [key for key in collector.data.keys() if key is not Topology]
+        # Sometimes it's good enough just to save in reverse deletion order.
+        duplicate_order = reversed(related_models)
+        for model in duplicate_order:
+            # Replace each `sub_obj` with a duplicate.
+            sub_objects = collector.data[model]
+            for obj in sub_objects:
+                # Duplicate the object and save it.
+                obj.id = None
+                obj.pk = None
+                obj.uuid = uuid.uuid4()
```

Comment:
```suggestion
                obj.uuid = None
```

I think that with default value uuid is automaticaly generated (to verify)